client/hostinfo_unix.cpp

```diff
@@ -1567,9 +1567,19 @@ inline long device_idle_time(const char *device) {
 static const struct dir_tty_dev {
     const char *dir;
     const char *dev;
+    const vector<string> ignore_list;
+
+    bool should_ignore(const std::string &devname) const {
+        for (auto ignore : ignore_list) {
```

Comment:
Yeah, I also like `auto` but please set here explicitly `const std::string &ignore`

Reply:
Out of curiosity (and cause I would prefer auto here): Why do you want to be explicit in this case about the type?

Reply:
There are couple of reasons:
1 - The code in our codebase is pretty old and written in one style so it's easier to read the code when even new code uses the same code style as the old one. In this particular case it's not a complicated template of something like this that is easier to write using `auto` and, by the way, even with `auto` you're missing `const` and `&`
2 - It's easier to do a code review in browser because you don't need to look what is the type is hidden behind `auto`